nbs/06_transform.ipynb

```diff
@@ -21,7 +21,16 @@
    "cell_type": "code",
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #59.            if lhs_node in rule.nodes_to_remove() or (lhs_node in cloned_to_flags_map.keys() and cloned_to_flags_map[lhs_node] == False):
</pre><p> </p><p> </p><p>instead of rule.nodes_to_remove() each time - we can factor it likes this:</p>

<pre class="ql-syntax" spellcheck="false">node_sets_to_remove = {
	collection_map(node)&nbsp;
	if node in collections
	else set(mapping[node])
	for node in nodes_to_remove
}
nodes_to_remove = set.union(*nodes_sets_to_remove)
</pre><p> </p><p>And use this set instead of the two different loops for lhs and collections</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/DeanLight/graph_rewrite/pull/17/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>